include/pagmo/problems/dtlz.hpp

```diff
@@ -0,0 +1,511 @@
+    * larger than an implementation defiend value
+    *
+    */
+    dtlz(unsigned int prob_id = 1u, vector_double::size_type dim = 5u, vector_double::size_type fdim = 3u,
+         unsigned int alpha = 100u)
+        : m_prob_id(prob_id), m_alpha(alpha), m_dim(dim), m_fdim(fdim)
+    {
+        if (prob_id == 0u || prob_id > 7u) {
+            pagmo_throw(std::invalid_argument, "DTLZ test suite contains seven (prob_id = [1 ... 7]) problems, prob_id="
+                                                   + std::to_string(prob_id) + " was detected");
+        }
+        if (fdim < 2u) {
+            pagmo_throw(std::invalid_argument, "DTLZ test problem have a minimum of 2 objectives: fdim="
+                                                   + std::to_string(fdim) + " was detected");
+        }
+        // We conservatively limit these dimensions to avoid checking overflows later
+        if (fdim > std::numeric_limits<decltype(fdim)>::max() / 3u) {
+            pagmo_throw(std::invalid_argument, "The number of objectives is too large");
+        }
+        if (dim > std::numeric_limits<decltype(dim)>::max() / 3u) {
+            pagmo_throw(std::invalid_argument, "The problem dimension is too large");
+        }
+        if (dim <= fdim) {
+            pagmo_throw(std::invalid_argument, "The problem dimension has to be larger than the number of objectives.");
+        }
+    }
+    /// Fitness computation
+    /**
+     * Computes the fitness for this UDP
+     *
+     * @param x the decision vector.
+     *
+     * @return the fitness of \p x.
+     */
+    vector_double fitness(const vector_double &x) const
+    {
+        vector_double retval;
+        switch (m_prob_id) {
+            case 1:
+                retval = f1_objfun_impl(x);
+                break;
+            case 2:
+            case 3:
+                retval = f23_objfun_impl(x);
+                break;
+            case 4:
+                retval = f4_objfun_impl(x);
+                break;
+            case 5:
+            case 6:
+                retval = f56_objfun_impl(x);
+                break;
+            case 7:
+                retval = f7_objfun_impl(x);
+                break;
+        }
+        return retval;
+    }
+    /// Number of objectives
+    /**
+     * One of the optional methods of any user-defined problem (UDP).
+     * It returns the number of objectives.
+     *
+     * @return the number of objectives
+     */
+    vector_double::size_type get_nobj() const
+    {
+        return m_fdim;
+    }
+    /// Box-bounds
+    /**
+     * One of the optional methods of any user-defined problem (UDP).
+     * It returns the box-bounds for this UDP.
+     *
+     * @return the lower and upper bounds for each of the decision vector components
+     */
+    std::pair<vector_double, vector_double> get_bounds() const
+    {
+        std::pair<vector_double, vector_double> retval{vector_double(m_dim, 0.), vector_double(m_dim, 1.)};
```

Comment:
Maybe compress on a single line with the init list syntax? ``return {vector_double(...), vector_double(...)};``.